### PR TITLE
Ignore name of index when fetching settings

### LINKED ISSUE #822: [BUG] Cannot update aliased index through Index.save()
### What is the bug?
I am unable to update the mapping of an index which is accessed by opensearch-py through an alias.

The relevant frame from the stack-trace is:
```
  File "/usr/local/lib/python3.10/site-packages/opensearchpy/helpers/index.py", line 328, in save
    current_settings = self.get_settings(using=using)[self._name]["settings"][
KeyError: 'alias_name'
```

### How can one reproduce the bug?
1. Create an index and alias pointing to the index
2. Create a `Document` (with any field) using the alias's name as the `Index.name` attribute
3. Run `.get_index().save()` on the created `Document` subclass

### What is the expected behavior?
I expect the underlying index to be updated through the alias, as is possible through direct requests.

### What is your host/environment?
Using opensearch-py 2.4.2 with CPython 3.10.

### Do you have any additional context?
I am willing to create a PR.

### PR DESCRIPTION
**Pull request by tienne-B**
### Description

This commit fixes a situation where an index cannot be updated through this client as the index is aliased, with the client pointing to the alias.

As the `GET /<index>/_settings` request will only ever return the settings for the specified index (through the alias), it would only have one key, so the name of the key would not matter. We can pop the key to get the settings object for the index through the alias.

_Describe what this change achieves._

### Issues Resolved

Closes #822 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tienne-B** [comment]
Thank you @dblock for your quick response! The value of removing `self._name` is that saving an index previously wouldn't work if OS-py accesses it through an alias (and using an alias seems to be a good practice for downtime prevention). All the requests that are triggered by the `.save()` method can work transparently through an index, however, when making the `GET /<alias>/_settings` request, the `dict` response key is the name of the true index, rather than the alias. This causes the `KeyError`. However, the data we're extracting doesn't depend on the name, so it'd be safe to disregard and pop the key/value.

If it's wanted, I could add a check that the alias only points to one index to raise a more specific error. Would that be appreciated?

I've added a test, but I'm a little skeptical of the coverage report as it's based off a 6-month-old commit.

Thank you!

**Comment by dblock** [comment]
> If it's wanted, I could add a check that the alias only points to one index to raise a more specific error. Would that be appreciated?

Possibly, but doesn't have to block this PR. Give it a shot after I merge this!

**Review by dblock — CHANGES_REQUESTED** [review]
This all makes sense.

I think you have to do the same for the async code (see https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/_async/helpers/index.py#L308). Add a test for that too.

**Comment by tienne-B** [comment]
I've added support for async, the checks for multiple indices, and testing for these.

File: opensearchpy/_async/helpers/index.py
Comment (dblock) [inline opensearchpy/_async/helpers/index.py:315]:
```suggestion
                "Settings for %s point to multiple indices: %s."
```

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good! 

I have some small language asks for the error. Also check that the tests do run because CodeCov seems unhappy and complains that tests are missing which seems odd.

File: opensearchpy/helpers/test.py
Comment (tienne-B) [reply opensearchpy/helpers/test.py:37]: Oh sorry, didn't mean to commit that! I had noticed that the tests did not run, and had to uncomment these list values to get the tests to run:

https://github.com/opensearch-project/opensearch-py/blob/a24b9f3f3cccd779ef0d9b1161ce8ce518717d10/test_opensearchpy/run_tests.py#L159-L164

I am a little hesitant to commit such changes as I don't have much context as to why this has been done and if it'd break anything. Also, the base commit for Codecov is quite old.

Comment (tienne-B) [reply opensearchpy/helpers/test.py:37]: I can assure you that they do pass locally with the files un-ignored.

On GitHub Actions they would be skipped as there is no OpenSearch service for the tests to use.

File: test_opensearchpy/test_async/test_server/test_helpers/test_index.py
Comment (tienne-B) [reply test_opensearchpy/test_async/test_server/test_helpers/test_index.py]: I'll use `assertRaises`; cleans it up a bit.

File: opensearchpy/helpers/test.py
Comment (dblock) [reply opensearchpy/helpers/test.py:37]: That's the whole point, integration tests should run on GHA in one of the workflows.

**Comment by tienne-B** [comment]
@dblock Rebased.

File: opensearchpy/helpers/test.py
Comment (dblock) [reply opensearchpy/helpers/test.py:37]: This is still here, do you want to revert this line?

**Review by dblock — CHANGES_REQUESTED** [review]
There's still that http, want to fix it?

File: opensearchpy/helpers/test.py
Comment (tienne-B) [reply opensearchpy/helpers/test.py]: Reverted, sorry.

**Review by dblock — APPROVED** [review]

**Pull request merged**
